tests/test_primary.py

```diff
@@ -523,20 +612,126 @@ def test_40_get_validated_target_info(self):
 
 
 
-  def test_45_get_image_for_ecu(self):
-    pass
+  def test_55_update_exists_for_ecu(self):
+
+    Registered_Unknown_Secondary = "potato" #Secondary that will be registered w/ primary as secondaries but will not listed by targets/director for any updates
+    Unregistered_Unknown_Secondary = "potato1" #Secondary that will be not registered w/ primary as secondaries and will not listed by targets/director for any updates
+    Registered_Known_Secondary = "TCUdemocar" #Secondary that will be registered w/ primary as secondaries and will be listed by targets/director for updates.
+    Registered_Unknown_Invalid_Secondary = 5 #Invalid ECU Serial for a secondary
+
+    # Registering valid names
+    TestPrimary.instance.register_new_secondary(Registered_Unknown_Secondary) 
+    TestPrimary.instance.register_new_secondary(Registered_Known_Secondary)
+
+    # Registering already registered names for testing lines in register_new_secondary()
+    TestPrimary.instance.register_new_secondary(Registered_Unknown_Secondary)
+
+    # Trying to register an invalid name
+    with self.assertRaises(tuf.FormatError):
+      TestPrimary.instance.register_new_secondary(Registered_Unknown_Invalid_Secondary)
+
+    #Asserting that as long as name is in a valid format it will be registered by the primary as a secondary.
+    self.assertIn(Registered_Unknown_Secondary, TestPrimary.instance.my_secondaries)
+    self.assertIn(Registered_Known_Secondary, TestPrimary.instance.my_secondaries)
+
+    with self.assertRaises(uptane.UnknownECU):
+      TestPrimary.instance._check_ecu_serial(Unregistered_Unknown_Secondary)
+
+    # Running a primary update cycle so it process all the files required for a establishing update cycle    
+    TestPrimary.instance.primary_update_cycle()
+
+    #Trying to get updates for an unregistered unknown ECU 
+    with self.assertRaises(uptane.UnknownECU):
+      TestPrimary.instance.update_exists_for_ecu(Unregistered_Unknown_Secondary)
+
+    #Trying to get updates for a registered secondary that is not listed by targets for updates
+    self.assertFalse(TestPrimary.instance.update_exists_for_ecu(Registered_Unknown_Secondary))
+
+    #Trying to get updates for a registered secondary that is listed by targets for updates
+    self.assertTrue(TestPrimary.instance.update_exists_for_ecu(Registered_Known_Secondary))
+
+    # delete pinned.json file because new pinned.json will be created depending on the current working directory of uptane every time the tests are run
+    #os.remove(TEST_TEMP_PINNING_FNAME)
+
```

Comment:
This function, test_55_update_exists_for_ecu I'll keep, in an edited squash commit so that it still attributes you.